libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/routeprogress/MetricsRouteProgress.kt

```diff
@@ -102,11 +103,19 @@ class MetricsRouteProgress(routeProgress: RouteProgress?) {
     }
 
     private fun obtainLegData(legProgress: RouteLegProgress) {
-        currentStepDistance = legProgress.currentStep().distance().toInt()
-        currentStepDuration = legProgress.currentStep().duration().toInt()
-        currentStepDistanceRemaining = legProgress.currentStepProgress().distanceRemaining().toInt()
-        currentStepDurationRemaining = legProgress.currentStepProgress().durationRemaining().toInt()
-        currentStepName = legProgress.currentStep().name() ?: ""
+        currentStepDistance = ifNonNull(legProgress.currentStep()?.distance()) { currentStepDistance ->
```

Comment:
Do we need to use `ifNonNull` here?
```suggestion
        currentStepDistance = legProgress.currentStep()?.distance()?.toInt() ?: 0
```
And the same for below

Reply:
Yes, we do. 
legProgress.currentStep()?.distance() returns null or Double, but we're assigning to currentStepDistance which is a nun-Nullable Double